compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java

```diff
@@ -344,62 +344,61 @@ private Map<String, PackageManifest.Platform> getPlatforms(TopLevelNode platform
-                                newPlatform = new PackageManifest.Platform(platforms.get(platformCode).dependencies(),
-                                        platforms.get(platformCode).repositories(),
-                                        newPlatform.graalvmCompatible());
+                for (String platformCode : platformCodes) {
+                    TopLevelNode platformCodeNode = platformTable.entries().get(platformCode);
+
+                    if (platformCodeNode.kind() == TomlType.TABLE) {
+                        TomlTableNode platformCodeTable = (TomlTableNode) platformCodeNode;
+                        // Get graalvmCompatible value
+                        TopLevelNode graalvmCompatibleNode = platformCodeTable.entries().get(GRAALVM_COMPATIBLE);
+                        if (graalvmCompatibleNode != null) {
+                            if (graalvmCompatibleNode.kind() == TomlType.NONE) {
+                                return platforms;
+                            }
+                            PackageManifest.Platform newPlatform =
+                                    getGraalvmCompatibilityPlatform(graalvmCompatibleNode);
+                            if (newPlatform != null) {
+                                if (platforms.get(platformCode) != null) {
+                                    newPlatform = new PackageManifest.Platform(platforms.
+                                            get(platformCode).dependencies(),
+                                            platforms.get(platformCode).repositories(),
+                                            newPlatform.graalvmCompatible());
+                                }
+                                platforms.put(platformCode, newPlatform);
                             }
-                            platforms.put(platformCode, newPlatform);
                         }
-                    }
 
-                    TopLevelNode topLevelNode = platformCodeTable.entries().get(DEPENDENCY);
-                    if (topLevelNode != null) {
-                        if (topLevelNode.kind() == TomlType.NONE) {
-                            return platforms;
-                        }
-                        PackageManifest.Platform newPlatform = getDependencyPlatform(topLevelNode);
-                        if (newPlatform != null) {
-                            if (platforms.get(platformCode) != null) {
-                                newPlatform = new PackageManifest.Platform(newPlatform.dependencies(),
-                                        platforms.get(platformCode).repositories(),
-                                        platforms.get(platformCode).graalvmCompatible());
+                        TopLevelNode topLevelNode = platformCodeTable.entries().get(DEPENDENCY);
+                        if (topLevelNode != null) {
+                            if (topLevelNode.kind() == TomlType.NONE) {
+                                return platforms;
```

Comment:
This would introduce a bug when we have two platforms and only the second platform has repositories. Shall we test and fix this too?